avro-fastserde/src/main/java/com/linkedin/avro/fastserde/FastSerdeCache.java

```diff
@@ -44,6 +44,21 @@
 
   private static volatile FastSerdeCache _INSTANCE;
 
+  /**
+   * Fast-avro will generate and load serializer and deserializer(SerDes) classes into metaspace during runtime.
+   * During serialization and deserialization, fast-avro also leverages JIT compilation to boost the SerDes speed.
+   * And JIT compilation code is saved in code cache.
+   * Too much usage of metaspace and code cache will bring GC/OOM issue.
+   *
+   * We set a hard limit of the total number of SerDes classes generated and loaded by fast-avro.
+   * By default, the limit is set to 400 based on our performance test with manually generated schemas.
+   * The number could keep the metaspace and codecache usage within a safe range.
+   * Fast-avro will fall back to regular avro after the limit is hit.
+   * One could get/set the limit by {@link #getSchemaNumLimit} and {@link #setSchemaNumLimit}
+   */
+  private static volatile int schemaNumLimit = 400;
```

Comment:
From my understanding, the goal is to limit the total number of the generated classes, so could we rename it to be `generatedFastClassNumLimit`?

Reply:
also - can we default this to max int? it would be better if people opted into this limit than if they find this out possibly weeks after deployment